lib/rubocop/cop/rails/schema_comment.rb

```diff
@@ -0,0 +1,120 @@
+      # to the database during a migration.
+      #
+      # @example
+      #   # bad (no comment for a new column or table)
+      #   add_column :table, :column, :integer
+      #
+      #   create_table :table do |t|
+      #     t.type :column
+      #   end
+      #
+      #   # good
+      #   add_column :table, :column, :integer, comment: 'Number of offenses'
+      #
+      #   create_table :table, comment: 'Table of offenses data' do |t|
+      #     t.type :column, comment: 'Number of offenses'
+      #   end
+      #
+      class SchemaComment < Base
+        include ActiveRecordMigrationsHelper
+
+        COLUMN_MSG = 'New database column without `comment`.'
+        TABLE_MSG = 'New database table without `comment`.'
+        RESTRICT_ON_SEND = %i[add_column create_table].freeze
+        CREATE_TABLE_COLUMN_METHODS = %w[
+          :references
+          :belongs_to
+          :primary_key
+          :column
+          :integer
+          :string
+          :text
+          :float
+          :decimal
+          :time
+          :date
+          :datetime
+          :timestamp
+          :binary
+          :boolean
+          :blob
+          :numeric
+        ].join(' | ').freeze
+
+        COMMENT_PRESENT_PATTERN = <<~PATTERN
+          <(hash <(pair {(sym :comment) (str "comment")} (_ [present?])) ...>) ...>
+        PATTERN
+
+        # @!method add_column?(node)
+        def_node_matcher :add_column?, <<~PATTERN
+          (send nil? :add_column _table _column _type _?)
+        PATTERN
+
+        # @!method add_column_with_comment?(node)
+        def_node_matcher :add_column_with_comment?, <<~PATTERN
+          (
+            send nil? :add_column _table _column _type #{COMMENT_PRESENT_PATTERN}
+          )
+        PATTERN
+
+        # @!method create_table?(node)
+        def_node_matcher :create_table?, <<~PATTERN
+          (
+            send nil? :create_table _table _?
+          )
+        PATTERN
+
+        # @!method create_table?(node)
+        def_node_matcher :create_table_with_comment?, <<~PATTERN
+          (
+            send nil? :create_table _table #{COMMENT_PRESENT_PATTERN} ...
+          )
+        PATTERN
+
+        # @!method t_column?(node)
+        def_node_matcher :t_column?, <<~PATTERN
+          (send _var {#{CREATE_TABLE_COLUMN_METHODS}} ...)
+        PATTERN
+
+        # @!method t_column_with_comment?(node)
+        def_node_matcher :t_column_with_comment?, <<~PATTERN
+          (send _var {#{CREATE_TABLE_COLUMN_METHODS}} _column _type? #{COMMENT_PRESENT_PATTERN})
+        PATTERN
+
+        def on_send(node)
+          if add_column_without_comment?(node)
+            add_offense(node, message: COLUMN_MSG)
+          elsif create_table?(node)
+            if create_table_without_comment?(node)
+              add_offense(node, message: TABLE_MSG)
+            elsif create_table_column_call_without_comment?(node)
+              add_offense(node.parent.body, message: COLUMN_MSG)
+            end
```

Comment:
@pirj I couldn't find a way to make a negation syntax.
I am checking what method was called and then checking if the comment is present.
I am doing this because of the mutiple `RESTRICT_ON_SEND`.
Do you know how to work around that and only making a `comment_present?` check?